torch/csrc/jit/mobile/module.cpp

```diff
@@ -46,14 +48,18 @@ namespace {
 void set_train_recurse(
     const c10::intrusive_ptr<c10::ivalue::Object>& obj,
     bool on) {
+  std::cout << "mobile: ivalue name: " << obj->name() << std::endl;
   if (auto slot = obj->type()->findAttributeSlot("training")) {
     obj->setSlot(*slot, on);
   } else {
     TORCH_INTERNAL_ASSERT(false, "'training' attribute not found");
   }
   for (const auto& slot : obj->slots()) {
-    if (slot.isObject()) {
-      set_train_recurse(slot.toObject(), on);
+    if (slot.isObject() && slot.toObjectRef().type()->is_module()) {
+      auto slot_obj = slot.toObject();
+      if (slot_obj->type()->hasAttribute("training")) {
```

Comment:
Is it possible that a module Parent does not have "training", but it's submodule Child has "training". Would it stop here at Parent and miss set_train for Child?

Reply:
Modify the logic in the latest commit. One thing I don't understand is the `TORCH_INTERNAL_ASSERT` part. Do we expect all modules have this attribute?